### PR TITLE
Handle .data pronoun in rename()

### PR DESCRIPTION
**Pull request by lionel-**

### CODE REVIEW

File: tests/testthat/test-select.r [hunk 1]
```diff
@@ -108,6 +108,10 @@ test_that("abort on unknown columns", {
   expect_error(select_vars(letters, c("a", "bar", "foo", "d")), "bar, foo")
 })
 
+test_that("rename() handles data pronoun", {
+  expect_identical(rename(tibble(x = 1), y = .data$x), tibble(y = 1))
```
Comment (krlmlr) [inline tests/testthat/test-select.r]: Just curious: Why is this necessary?

Comment (lionel-) [inline tests/testthat/test-select.r]: That's because it is our official way of avoiding a R CMD check note about unknown global variables.

Comment (krlmlr) [inline tests/testthat/test-select.r]: I see. Does `y = sym("x")` also work?

Comment (lionel-) [inline tests/testthat/test-select.r]: not currently, you'd have to unquote it. We'll have to wait until 0.7 for a tidyeval implementation of rename() :/

The difficulty is that we need to ignore evaluation errors about unbinded symbols when `strict = FALSE`.

Comment (hadley) [inline tests/testthat/test-select.r]: Consistency with the other verbs + avoid R CMD check note
